trie/database.go

```diff
@@ -847,3 +852,46 @@ func (db *Database) Size() (common.StorageSize, common.StorageSize) {
 	var metarootRefs = common.StorageSize(len(db.dirties[common.Hash{}].children) * (common.HashLength + 2))
 	return db.dirtiesSize + db.childrenSize + metadataSize - metarootRefs, db.preimagesSize
 }
+
+func (db *Database) saveCache(dir string, threads int) (err error) {
+	if db.cleans == nil {
+		return
+	}
+	defer func(start time.Time) {
+		if err == nil {
+			log.Info("Saved clean cache into the file", "path", dir, "elapsed", common.PrettyDuration(time.Since(start)))
+		} else {
+			log.Info("Failed to save clean cache into file", "error", err)
+		}
+	}(time.Now())
```

Comment:
This defer seems a bit forced. Lets just do start := time.Now() and have the err check + logs after `err = db.cleans.SaveToFileConcurrent(dir, threads)`. There's really no need to defer this, just makes things less readable.